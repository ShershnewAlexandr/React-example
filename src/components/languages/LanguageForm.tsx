import React, { Fragment } from 'react';
import { Field } from 'redux-form/immutable';
import { Grid } from '@material-ui/core';
import NumberInputComponent from '../inputs/NumberInput';
import { Language } from '../../types/formDataTypes';
import { LanguageFormContainerProps } from '../../containers/languages/LanguageFormContainer';

function LanguageForm(props: LanguageFormContainerProps) {
  const { languagesList, handleSubmit, isLoading } = props;
  const languagesElements = languagesList.map((current: Language) => (
    <Fragment key={current.native_name}>
      <Grid item xs={6}>
        <Field
          name={current.code}
          component={NumberInputComponent}
          label={current.native_name}
          min={0}
          max={5}
          step={1}
          disabled={isLoading}
        />
      </Grid>
    </Fragment>
  ));

  return (
    <form onSubmit={handleSubmit} data-testid="languagesForm">
      <Grid container direction="row" spacing={4}>
        {languagesElements}
      </Grid>
    </form>
  );
}

LanguageForm.defaultProps = {
  languagesList: [],
};

export default LanguageForm;
