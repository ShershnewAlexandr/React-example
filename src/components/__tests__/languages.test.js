import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import LanguageForm from '../languages/LanguageForm';

describe('languagesForm', () => {
  const props = {
    languagesList: [
      { native_name: 'name', code: 'ru' },
      { native_name: 'name', code: 'ru' },
    ],
    handleSubmit: jest.fn(),
    isLoading: false,
  };

  const languageForm = shallow(<LanguageForm {...props} />);

  it('snapshot', () => {
    expect(toJson(languageForm)).toMatchSnapshot();
  });

  it('should catch submit', () => {
    languageForm.find('form').simulate('submit');
    expect(props.handleSubmit.mock.calls.length).toBe(1);
  });

  it('have 2 field', () => {
    expect(languageForm.find('Field')).toHaveLength(2);
  });

  it('have default languagesList = []', () => {
    const languageForm = shallow(<LanguageForm />);
    expect(languageForm.find('Field')).toHaveLength(0);
  });
});
