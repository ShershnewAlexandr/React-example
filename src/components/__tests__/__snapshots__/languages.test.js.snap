// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languagesForm snapshot 1`] = `
<form
  data-testid="languagesForm"
  onSubmit={[MockFunction]}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="row"
    spacing={4}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={6}
    >
      <Field
        component={[Function]}
        disabled={false}
        label="name"
        max={5}
        min={0}
        name="ru"
        step={1}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={6}
    >
      <Field
        component={[Function]}
        disabled={false}
        label="name"
        max={5}
        min={0}
        name="ru"
        step={1}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</form>
`;
